RELEASE.md

```diff
@@ -76,9 +76,9 @@ manual release was performed.
 1. If there are any 'next' branches in GitHub, say for a future major version of Stencil (e.g. `v5.0.0-dev`), now is a
    good time to rebase them against the `main` branch.
 1. Perform the following tasks in JIRA:
-   1. Mark this version of Stencil as 'released' in JIRA on the 'Releases' page.
+   1. Ask someone with appropriate permissions to mark this version of Stencil as 'released' in JIRA on the 'Releases' page.
    1. Move the task card in this current sprint to the 'Done' swim-lane.
```

Comment:
We haven't created this card in a while. Thoughts on just removing this step?

Reply:
sure, fine by me, I think if we already have the 'release' entity in JIRA no need for a card as well